src/GoTrueClient.ts

```diff
@@ -89,33 +85,30 @@ export default class GoTrueClient {
    * Create a new client for use in the browser.
    * @param options.url The URL of the GoTrue server.
    * @param options.headers Any additional headers to send to the GoTrue server.
-   * @param options.storageKey Optional key name used for storing tokens in local storage
    * @param options.detectSessionInUrl Set to "true" if you want to automatically detects OAuth grants in the URL and signs in the user.
    * @param options.autoRefreshToken Set to "true" if you want to automatically refresh the token before expiring.
    * @param options.persistSession Set to "true" if you want to automatically save the user session into local storage. If set to false, session will just be saved in memory.
-   * @param options.localStorage Provide your own local storage implementation to use instead of the browser's local storage.
+   * @param options.storage Provide your own storage implementation to use instead of the browser's local storage.
    * @param options.multiTab Set to "false" if you want to disable multi-tab/window events.
    * @param options.cookieOptions
    * @param options.fetch A custom fetch implementation.
    */
   constructor(options: {
     url?: string
     headers?: { [key: string]: string }
-    storageKey?: string
     detectSessionInUrl?: boolean
     autoRefreshToken?: boolean
     persistSession?: boolean
-    localStorage?: SupportedStorage
+    storage?: SupportedStorage
     multiTab?: boolean
     cookieOptions?: CookieOptions
     fetch?: Fetch
   }) {
     const settings = { ...DEFAULT_OPTIONS, ...options }
     this.inMemorySession = null
-    this.storageKey = settings.storageKey
     this.autoRefreshToken = settings.autoRefreshToken
     this.persistSession = settings.persistSession
-    this.localStorage = settings.localStorage || globalThis.localStorage
+    this.storage = new LocalStorage({ key: PERSISTENCE_KEY })
```

Comment:
shouldn't you take this from the constructor and use `LocalStorage` as the default?